dynamo-store.js

```diff
@@ -590,12 +613,15 @@ function make_intern() {
             listreq.ExpressionAttributeNames = {}
             listreq.ExpressionAttributeNames[`#${pk}n`] = pk
 
+        
+        
             delete fq[pk]
 
             let sk = indexdefkey.sort
             if (null != sk && null != fq[sk]) {
-              listreq.KeyConditionExpression += ` and #${sk}n = :${sk}i`
-              listreq.ExpressionAttributeValues[`:${sk}i`] = fq[sk]
+              let fq_op = get_op(fq[sk])
+              listreq.KeyConditionExpression += null == fq_op ? ` and #${sk}n = :${sk}i` : ` and #${sk}n ${fq_op.op} :${sk}i`
```

Comment:
I am no DynamoDB user but this looks like this may be a security vulnerability. Usually, when you concatenate strings onto a db query, - unless you are the one who sets them - you must escape such strings. E.g. SQL injection vulnerability.

Please confirm that this poses no security vulnerability when used against DynamoDB.

Reply:
Speaking of SQL injection, isn't DynamoDB a NoSQL database? I wouldn't know of any other security vulnerabilities when concentrating strings onto a query. Will need to research. But good point though!

Reply:
DynamoDB is a NoSQL database, but the idea is the same - if you are injecting an unsanitized string into a query that is also a string - you got an injection vulnerability.

Reply:
where does each element come from? `sk`, `fq_op.op`, etc?
are they user provided?

Reply:
yes - user-provided. If you look at the tests: https://github.com/senecajs/seneca-dynamo-store/blob/71f892a7491319ddfdedb975f3996ab7dd4b79ed/test/dynamo-store.test.js#L278

sk - key
fq_op.op is now fq_op.cmpop as it goes for a comparison operator - also user-provided